rdkPlugins/OOMCrash/source/OOMCrashPlugin.cpp

```diff
@@ -0,0 +1,246 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "OOMCrashPlugin.h"
+/**
+ * Need to do this at the start of every plugin to make sure the correct
+ * C methods are visible to allow PluginLauncher to find the plugin
+ */
+REGISTER_RDK_PLUGIN(OOMCrash);
+
+/**
+ * @brief Constructor - called when plugin is loaded by PluginLauncher
+ *
+ * Do not change the parameters for this constructor - must match C methods
+ * created by REGISTER_RDK_PLUGIN macro
+ *
+ * Note plugin name is not case sensitive
+ */
+OOMCrash::OOMCrash(std::shared_ptr<rt_dobby_schema> &containerConfig,
+                             const std::shared_ptr<DobbyRdkPluginUtils> &utils,
+                             const std::string &rootfsPath)
+    : mName("OOMCrash"),
+      mContainerConfig(containerConfig),
+      mRootfsPath(rootfsPath),
+      mUtils(utils)
+{
+    AI_LOG_FN_ENTRY();
+
+    AI_LOG_FN_EXIT();
+}
+
+/**
+ * @brief Set the bit flags for which hooks we're going to use
+ *
+ * This plugin uses the postInstallation and postHalt hooks, so set those flags
+ */
+unsigned OOMCrash::hookHints() const
+{
+    return (
+        IDobbyRdkPlugin::HintFlags::PostInstallationFlag |
+	IDobbyRdkPlugin::HintFlags::PostHaltFlag);
+}
+
+/**
+ *  * @brief Dobby Hook - run in host namespace *once* when container bundle is downloaded
+ *   */
+bool OOMCrash::postInstallation()
+{
+    if (!mContainerConfig)
+    {
+        AI_LOG_WARN("Container config is null");
+        return false;
+    }
+
+    const std::string path = mContainerConfig->rdk_plugins->oomcrash->data->path;
+    if (!mUtils->mkdirRecursive(mRootfsPath + path.c_str(), 0755) && errno != EEXIST)
+    {
+        AI_LOG_ERROR("failed to create directory '%s' (%d - %s)", (mRootfsPath + path).c_str(), errno, strerror(errno));
+        return false;
+    }
+
+    if (!mUtils->mkdirRecursive(path.c_str(), 0755) && errno != EEXIST)
+    {
+        AI_LOG_ERROR("failed to create directory '%s' (%d - %s)", path.c_str(), errno, strerror(errno));
+        return false;
+    }
+
+    if (!mUtils->addMount(path, path, "bind", {"bind", "ro", "nodev","nosuid", "noexec" }))
+    {
+        AI_LOG_WARN("failed to add mount %s", path.c_str());
+        return false;
+    }
+    AI_LOG_INFO("OOMCrash postInstallation hook is running for container with hostname %s", mUtils->getContainerId().c_str());
+    return true;
+}
+
+/**
+ * @brief Dobby Hook - Run in host namespace when container terminates
+ */
+bool OOMCrash::postHalt()
+{
+    if (!mContainerConfig)
+    {
+        AI_LOG_WARN("Container config is null");
+        return false;
+    }
+    
+    bool status = true;
+    if (mUtils->exitStatus != 0)
+        status = checkForOOM();
+
+    if (status == true)
+        createFileForOOM();
+    
+    // Remove the crashFile if container exits normally or if no OOM detected
+    if (mUtils->exitStatus == 0 || status == false)
+    {
+	struct stat buffer;
```

Comment:
Wrong indentation level here. The best would be to set in your VSCode option "Replace tabs with spaces" and set "Tab size" to "4" as it is done in whole this project. This way you could tell by yourself that there was a mistake here.